### PR TITLE
26 enhance index

### PR DESCRIPTION
**Pull request by dikastes**

### CODE REVIEW

File: Classes/Processing/BibEntryProcessor.php [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        Collection $teiDataSets
+    ): array
+    {
+        $key = $bibliographyItem['key'];
+        $bibliographyItem['localizedCitations'] = [];
+        foreach ($localizedCitations as $locale => $localizedCitation) {
+            $bibliographyItem['localizedCitations'][$locale] = $localizedCitation->get($key)['citation'];
+        }
+        $bibliographyItem['tei'] = $teiDataSets->get($key);
+        $bibliographyItem['tx_lisztcommon_header'] = self::buildListingField($bibliographyItem, BibEntryConfig::HEADER_FIELDS);
```
Comment (thomas-sc) [inline Classes/Processing/BibEntryProcessor.php:35]: maybe standard names over all search indexes are better?

File: composer.json [hunk 2]
```diff
@@ -12,7 +12,7 @@
         "dikastes/zotero-api": "^1.2",
         "elasticsearch/elasticsearch": "^8",
         "illuminate/collections": "^11",
-        "slub/liszt-common": "@dev"
+        "slub/liszt-common": "^1.2"
```
Comment (thomas-sc) [inline composer.json:15]: composer update only runs on my local machine if i leave @dev here

**Review by thomas-sc — DISMISSED** [review]
short review ;-) 
the index -a command ran through in my test

File: Classes/Command/IndexCommand.php [hunk 3]
```diff
@@ -333,15 +335,15 @@ protected function buildDataSets(): void
     {
         $this->dataSets = $this->bibliographyItems->
             map(function($bibliographyItem) {
-                return self::buildDataSet($bibliographyItem, $this->localizedCitations, $this->teiDataSets);
+                return BibEntryProcessor::process($bibliographyItem, $this->localizedCitations, $this->teiDataSets);
             });
     }
 
     protected static function buildDataSet(
```
[the remark this answers is not in the document]
Comment (dikastes) [reply Classes/Command/IndexCommand.php:342]: right!

**Pull request merged**